packages/preferences/src/browser/preference-tree-provider.ts

```diff
@@ -0,0 +1,140 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, postConstruct } from 'inversify';
+import * as fuzzy from 'fuzzy';
+import { debounce } from 'lodash';
+import { TreeNode, CompositeTreeNode, PreferenceSchemaProvider, PreferenceDataSchema, PreferenceDataProperty, PreferenceScope } from '@theia/core/lib/browser';
+import { PreferenceConfigurations } from '@theia/core/lib/browser/preferences/preference-configurations';
+import { PreferencesEventService } from './util/preference-event-service';
+import { PreferenceTreeGenerator } from './util/preference-tree-generator';
+import { Preference } from './util/preference-types';
+
+interface PreferenceFilterOptions {
+    minLength?: number;
+    baseSchemaAltered?: boolean;
+};
+
+const filterDefaults: Required<PreferenceFilterOptions> = {
+    minLength: 1,
+    baseSchemaAltered: false,
+};
+
+@injectable()
+export class PreferencesTreeProvider {
+
+    protected lastSearchedLiteral: string = '';
+    protected lastSearchedFuzzy: string = '';
+    protected baseSchema: PreferenceDataSchema;
+    protected baseTree: CompositeTreeNode;
+    protected _currentTree: CompositeTreeNode;
+    protected currentScope: Preference.SelectedScopeDetails = Preference.DEFAULT_SCOPE;
+    protected handleUnderlyingDataChange = debounce(
+        (options: PreferenceFilterOptions, newScope?: Preference.SelectedScopeDetails) => this.updateUnderlyingData(options, newScope),
+        200
+    );
+    protected handleSearchChange = debounce((term: string) => this.updateDisplay(term), 100);
```

Comment:
unused?

Reply:
Yes, this ended up going unused. Instead, the search event fire is debounced. Shall we remove it now before this gets merged?